### PR TITLE
MAUI-2501: UG document for Maui Maps imagery layer

### PR DESCRIPTION
**Pull request by Shalini-Ashokan**
Hi @vigneshrameshsync 
I have prepared UG document for Maui Maps imagery layer. Could you please review it and if you have any concerns?

Regards,
Shalini Ashokan.

### CODE REVIEW

File: MAUI/Maps/tile-layer.md [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+map.Layer = tileLayer;
+this.Content = map;
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Clear a cached tile images from application memory
+
+The `DeleteTilesFromCache` method used to clear the cached tile images from  application cache memory.
```
Comment (florenceliliana) [inline MAUI/Maps/tile-layer.md:190]: The `DeleteTilesFromCache` method is used to clear the cached tile images from the application cache memory.

File: MAUI/Maps/tile-layer.md [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+
+tileLayer.DeleteTilesFromCache();
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Markers
+
+You can add markers in the tile layer. The procedure is very similar to the shape layer. Kindly refer the [markers](https://help.syncfusion.com/maui/maps/markers) section.
```
Comment (florenceliliana) [inline MAUI/Maps/tile-layer.md:214]: You can add markers in the tile layer. The procedure is very similar to the shape layer. Kindly refer to the [markers](https://help.syncfusion.com/maui/maps/markers) section.

File: MAUI/Maps/tile-layer.md [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+N> You can refer to our [.NET MAUI Maps](https://www.syncfusion.com/maui-controls/maui-maps) feature tour page for its groundbreaking feature representations. You can also explore our `.NET MAUI Maps Tile layer example` that shows how to configure a Maps in .NET MAUI.
```
Comment (florenceliliana) [inline MAUI/Maps/tile-layer.md:216]: N> You can refer to our [.NET MAUI Maps](https://www.syncfusion.com/maui-controls/maui-maps) feature tour page for its groundbreaking feature representations. You can also explore our `.NET MAUI Maps Tile layer example,` which shows how to configure Maps in .NET MAUI.

File: MAUI/Maps/zoom-pan.md [hunk 4]
```diff
@@ -0,0 +1,633 @@
...
+title: Zooming and Panning  in Maps control | Syncfusion
+description: Learn here all about the Shape Sublayer feature of the Syncfusion .NET MAUI Maps (SfMaps) control to customize their appearances and more.
+platform: MAUI
+control: SfMaps
+documentation: ug
+---
+
+# Zooming and Panning in .NET MAUI Maps (SfMaps)
+
+It is possible to zoom in and out for any layer to take a closer look at a specific region by pinching the map, scrolling the mouse wheel or track pad. Pan the map to navigate across the regions. You can also customize the zoom level of the rendering.
```
Comment (florenceliliana) [inline MAUI/Maps/zoom-pan.md:12]: It is possible to zoom in and out for any layer to take a closer look at a specific region by pinching the map or scrolling the mouse wheel or trackpad. Pan the map to navigate across the regions. You can also customize the zoom level of the rendering.

File: MAUI/Maps/zoom-pan.md [hunk 5]
```diff
@@ -0,0 +1,633 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Osm map default](images/zoom-pan/osm_map.png)
+
+## Customizing the zoom level
+
+You can set the zoom level of the map layer by using `MapZoomPanBehavior.ZoomLevel` property.
```
Comment (florenceliliana) [inline MAUI/Maps/zoom-pan.md:86]: You can set the zoom level of the map layer by using the `MapZoomPanBehavior.ZoomLevel` property.

File: MAUI/Maps/zoom-pan.md [hunk 6]
```diff
@@ -0,0 +1,633 @@
...
+
+The default `MapZoomPanBehavior.ZoomLevel` value is 1 which will show the whole map in the viewport for `MapTileLayer`.
```
Comment (florenceliliana) [inline MAUI/Maps/zoom-pan.md:88]: The default `MapZoomPanBehavior.ZoomLevel` value is 1, which will show the whole map in the viewport for `MapTileLayer.`

File: MAUI/Maps/zoom-pan.md [hunk 7]
```diff
@@ -0,0 +1,633 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![OSM map zoomlevel](images/zoom-pan/osm_maps_zoomlevel.png)
+
+## Enable zooming 
+
+The zooming feature enables you zoom in and zoom out the maps to show the in-depth information. To enable zooming, set the `EnableZooming` property to `true`. The default value of the `EnableZooming` is `true`.
```
Comment (florenceliliana) [inline MAUI/Maps/zoom-pan.md:131]: The zooming feature enables you to zoom in and zoom out the maps to show in-depth information. To enable zooming, set the `EnableZooming` property to `true.` The default value of the `EnableZooming` is `true.`

File: MAUI/Maps/zoom-pan.md [hunk 8]
```diff
@@ -0,0 +1,633 @@
...
+map.Layer = tileLayer;
+this.Content = map;
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Enable Panning 
+
+Panning feature allows you move the visible area of the maps when it is zoomed in. To enable panning, set the `EnablePanning` property to `true`. The default value of the `EnablePanning` is `true`.
```
Comment (florenceliliana) [inline MAUI/Maps/zoom-pan.md:167]: The panning feature allows you to move the visible area of the maps when it is zoomed in. To enable panning, set the `EnablePanning` property to `true.` The default value of the `EnablePanning` is `true.`

File: MAUI/Maps/zoom-pan.md [hunk 9]
```diff
@@ -0,0 +1,633 @@
...
+map.Layer = tileLayer;
+this.Content = map;
+
+{% endhighlight %}
+
+{% endtabs %}
+
+## Customizing min and max zoom level
+
+You can set the min and max zoom level of the map layer by setting the value to `MapZoomPanBehavior.MinZoomLevel` and `MapZoomPanBehavior.MaxZoomLevel` properties. The minimum and maximum zooming levels can be restricted using these properties respectively. The default values of `MapZoomPanBehavior.MinZoomLevel` and `MapZoomPanBehavior.MaxZoomLevel` are 1 and 15 respectively.
```
Comment (florenceliliana) [inline MAUI/Maps/zoom-pan.md:205]: You can set the min and max zoom level of the map layer by setting the value to `MapZoomPanBehavior.MinZoomLevel` and `MapZoomPanBehavior.MaxZoomLevel` properties. The minimum and maximum zooming levels can be restricted using these properties, respectively. The default values of `MapZoomPanBehavior.MinZoomLevel` and `MapZoomPanBehavior.MaxZoomLevel` are 1 and 15, respectively.

File: MAUI/Maps/zoom-pan.md [hunk 10]
```diff
@@ -0,0 +1,633 @@
...
+
+However, for `MapTileLayer`, `MapZoomPanBehavior.maxZoomLevel` may slightly vary depends on the providers. Kindly check the respective official website of the map tile providers to know about the maximum zoom level it supports.
```
Comment (florenceliliana) [inline MAUI/Maps/zoom-pan.md:207]: However, for `MapTileLayer,` the `MapZoomPanBehavior.maxZoomLevel` may vary slightly depending on the providers. Kindly check the respective official website of the map tile providers to know about the maximum zoom level it supports.

File: MAUI/Maps/zoom-pan.md [hunk 11]
```diff
@@ -0,0 +1,633 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![SfMaps zoom level changed image](images/zoom-pan/distance_radius.png)
+
+### Geo-bounds
+
+Calculate the zoom level automatically based on the `MapLatLngBounds(Northeast, Southwest)` of `MapTileLayer` class.
```
Comment (florenceliliana) [inline MAUI/Maps/zoom-pan.md:367]: Calculate the zoom level automatically based on the `MapLatLngBounds(Northeast, Southwest)` of the `MapTileLayer` class.

File: MAUI/Maps/zoom-pan.md [hunk 12]
```diff
@@ -0,0 +1,633 @@
...
+
+{% endtabs %}
+
+![Double tap zooming](images/zoom-pan/double_tap_zoom.gif)
+
+## Animation
+
+### Zoom level Animation
+
+To perform animation on zoom level changing using the `MapTileLayer.EnableZoomingAnimation` property. The default value of the `EnableZoomingAnimation` is `true`.
```
Comment (florenceliliana) [inline MAUI/Maps/zoom-pan.md:529]: To perform the animation on zoom level changing using the `MapTileLayer.EnableZoomingAnimation` property. The default value of the `EnableZoomingAnimation` is `true.`

File: MAUI/Maps/zoom-pan.md [hunk 13]
```diff
@@ -0,0 +1,633 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Zoom animation](images/zoom-pan/zoom_animation.gif)
+
+### Center Animation
+
+To perform animation on center value changing using the `MapTileLayer.EnableCenterAnimation` property. The default value of the `EnableCenterAnimation` is `true`.
```
Comment (florenceliliana) [inline MAUI/Maps/zoom-pan.md:564]: To perform the animation on center value changing using the `MapTileLayer.EnableCenterAnimation` property. The default value of the `EnableCenterAnimation` is `true.`

File: MAUI/Maps/zoom-pan.md [hunk 14]
```diff
@@ -0,0 +1,633 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Center animation](images/zoom-pan/center_zoom_animation.gif)
+
+## Event
+
+The `ZoomLevelChanging` event triggers when zoom level changed. Following arguments can be get from the ZoomLevelChanging event .
```
Comment (florenceliliana) [inline MAUI/Maps/zoom-pan.md:599]: The `ZoomLevelChanging` event triggers when the zoom level is changed. Following arguments can be get from the ZoomLevelChanging event.

**Review by florenceliliana — CHANGES_REQUESTED** [review]

File: MAUI/Maps/tile-layer.md [hunk 15]
```diff
@@ -0,0 +1,216 @@
...
+documentation: ug
+---
+
+# Tile Layer in in .NET MAUI Maps (SfMaps)
+
+The tile layer renders the tiles returned from web map tile services such as Bing Maps, OpenStreetMaps, Google Maps, TomTom, etc.
+
+## Setting URL template
+
+The `MapTileLayer` needs to be added in the `Layer` collection in `SfMaps`. The URL of the providers must be set in the `MapTileLayer.UrlTemplate` property.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified

File: MAUI/Maps/tile-layer.md [hunk 16]
```diff
@@ -0,0 +1,216 @@
...
+
+The `UrlTemplate` property accepts the URL in [`WMTS`](https://en.wikipedia.org/wiki/Web_Map_Tile_Service) format i.e. {z} — zoom level, {x} and {y} — tile coordinates. This URL might vary slightly depends on the providers. The formats can be,
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Updated

File: MAUI/Maps/tile-layer.md [hunk 17]
```diff
@@ -0,0 +1,216 @@
...
+    https://example_provider/{z}/{x}/{y}.png,
+    https://example_provider/z={z}/x={x}/y={y}.png,
+    https://example_provider/z={z}/x={x}/y={y}.png?key=subscription_key, etc.
+
+We will replace the {z}, {x}, {y} internally based on the current `Center` and the zoom level.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Changed

File: MAUI/Maps/tile-layer.md [hunk 18]
```diff
@@ -0,0 +1,216 @@
...
+
+N> Some of the providers may need subscription key. Please include them in the `UrlTemplate` itself as mentioned in the above example. Please note that the format may vary between each map providers. You can check the exact URL format needed for the providers on their official websites.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified the content

File: MAUI/Maps/tile-layer.md [hunk 19]
```diff
@@ -0,0 +1,216 @@
...
+
+## Adding OSM/OpenStreetMap
+
+The OpenStreetMap is one of the tile/image providers which can be used free of cost. It returns map tiles for the requested coordinates for every requests. The url format of the OSM map provider as shown in the below code snippet.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified the content

File: MAUI/Maps/zoom-pan.md (hunk 11 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Modified the content

File: MAUI/Maps/zoom-pan.md (hunk 12 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Modified the content

File: MAUI/Maps/zoom-pan.md (hunk 13 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Modified the content

File: MAUI/Maps/zoom-pan.md (hunk 14 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Changed the content

File: MAUI/Maps/zoom-pan.md [hunk 20]
```diff
@@ -0,0 +1,633 @@
...
+        e.Cancel = true; // Cancels the zooming event
+        var NewZoomLevel = e.NewZoomLevel; // Returns the current zoomed level
+    }
+}
+
+{% endhighlight %}
+
+{% endtabs %}
+
+N> You can refer to our [.NET MAUI Maps](https://www.syncfusion.com/maui-controls/maui-maps) feature tour page for its groundbreaking feature representations. You can also explore our `.NET MAUI Maps Zoom & Pan example` that shows how to configure a Maps in .NET MAUI.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Modified the content

File: MAUI/Maps/tile-layer.md [hunk 21]
```diff
@@ -0,0 +1,216 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![OSM default view](images/tile-layer/osm_maps_default.png)
+
+## Adding Bing maps
+
+An additional step is required for the Bing maps. The format of the required URL varies from the other tile services. Hence, we have added a top-level `GetTileUrl` method which returns the URL in the required format. The subscription key is needed for bing maps. You can create an API key by following the steps mentioned in this [`link`](https://docs.microsoft.com/en-us/bingmaps/getting-started/bing-maps-dev-center-help/getting-a-bing-maps-key) and append this key to the bing map url before pass it to the `GetTileUrl` method. You can use the URL returned from this method to pass it to the `UrlTemplate` property.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified the content

File: MAUI/Maps/tile-layer.md [hunk 22]
```diff
@@ -0,0 +1,216 @@
...
+
+Some of the providers provide different map types. For example, Bing Maps provide map types like Road, Aerial, AerialWithLabels etc. These types too can be passed in the `UrlTemplate` itself as shown in the below example. You can check the official websites of the tile providers to know about the available types and the code for it.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified the content

File: MAUI/Maps/tile-layer.md [hunk 23]
```diff
@@ -0,0 +1,216 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![Bing maps default view](images/tile-layer/bing_maps_default.png)
+
+## Other map tile providers
+
+Our tile layer is not limited or specific to any of the tile providers mentioned here. It supports requesting tiles from any of the tile providers using the unique URL for respective tile providers and renders them.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified the content

File: MAUI/Maps/tile-layer.md [hunk 24]
```diff
@@ -0,0 +1,216 @@
...
+
+For other map providers like TomTom, MapBox etc., you can check the respective official websites and provide the url in the format mentioned in the `Setting URL template` section.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified the content

File: MAUI/Maps/tile-layer.md [hunk 25]
```diff
@@ -0,0 +1,216 @@
...
+
+{% endhighlight %}
+
+{% endtabs %}
+
+![TomTom maps default view](images/tile-layer/tom_tom_default.png)
+
+## Changing the center latitude and longitude
+
+You can set the center position by setting the `MapTileLayer.Center` property. It represents the center position of the map layer.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md:112]: Modified the content

File: MAUI/Maps/tile-layer.md [hunk 26]
```diff
@@ -0,0 +1,216 @@
...
+
+Based on the size of the [`SfMaps`](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.Maps.html) control, `Center` and `ZoomLevel` number of initial tiles needed in the view port alone will be rendered. Refer this section for enabling `zooming and panning`.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified the content

File: MAUI/Maps/tile-layer.md [hunk 27]
```diff
@@ -0,0 +1,216 @@
...
+
+{% endtabs %}
+
+![OSM map center](images/tile-layer/osm_center.png)
+
+## Cache a tile images in application memory
+
+The `CanCacheTiles` property used to decide whether the tile images should be cached in application memory or not. The default value of the `CanCacheTiles` is `false`.
+
+While enabling the `CanCacheTiles` as `true` then need to set the tile server name to maintain the folder to store cache tiles in `MapTileLayer.UrlTemplate` property. The default tile server name to store tile cache is OSM. If you want to change server name you can use the below URL format.
```
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified the content

File: MAUI/Maps/tile-layer.md (hunk 1 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified the content

File: MAUI/Maps/tile-layer.md (hunk 2 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified the content

File: MAUI/Maps/tile-layer.md (hunk 3 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Modified the content

File: MAUI/Maps/zoom-pan.md (hunk 4 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Changed the content

File: MAUI/Maps/zoom-pan.md (hunk 5 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Changed the content

File: MAUI/Maps/zoom-pan.md (hunk 6 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Changed the content

File: MAUI/Maps/zoom-pan.md (hunk 7 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Changed the content

File: MAUI/Maps/zoom-pan.md (hunk 8 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Updated the content

File: MAUI/Maps/zoom-pan.md (hunk 9 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Updated the content

File: MAUI/Maps/zoom-pan.md (hunk 10 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Updated the content

File: MAUI/Maps/tile-layer.md [hunk 28]
```diff
@@ -0,0 +1,216 @@
...
+documentation: ug
+---
+
+# Tile Layer in in .NET MAUI Maps (SfMaps)
+
+The tile layer renders the tiles returned from web map tile services such as Bing Maps, OpenStreetMaps, Google Maps, TomTom, etc.
+
+## Setting URL template
+
+The `MapTileLayer` needs to be added to the `Layer` collection in `SfMaps`. The URL of the providers must be set in the `MapTileLayer.UrlTemplate` property.
```
Comment (vigneshrameshsync) [inline MAUI/Maps/tile-layer.md:16]: Whether `Layer` is a collection type?

File: MAUI/Maps/tile-layer.md (hunk 24 above)
[the remark this answers is not in the document]
Comment (vigneshrameshsync) [reply MAUI/Maps/tile-layer.md]: @Shalini-Ashokan Comma missing after MapBox

File: MAUI/Maps/tile-layer.md (hunk 27 above)
[the remark this answers is not in the document]
Comment (vigneshrameshsync) [reply MAUI/Maps/tile-layer.md]: @Shalini-Ashokan  `the` missing in 2 places.

File: MAUI/Maps/zoom-pan.md (hunk 4 above)
Comment (vigneshrameshsync) [reply MAUI/Maps/zoom-pan.md]: @Shalini-Ashokan `trackpad` is single word.

File: MAUI/Maps/zoom-pan.md (hunk 14 above)
Comment (vigneshrameshsync) [reply MAUI/Maps/zoom-pan.md]: Extra space before `.`

File: MAUI/Maps/tile-layer.md (hunk 17 above)
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Updated

File: MAUI/Maps/tile-layer.md (hunk 24 above)
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Updated

File: MAUI/Maps/tile-layer.md (hunk 27 above)
[the remark this answers is not in the document]
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md]: Updated

File: MAUI/Maps/zoom-pan.md (hunk 4 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Changed

File: MAUI/Maps/zoom-pan.md (hunk 14 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/zoom-pan.md]: Removed

File: MAUI/Maps/tile-layer.md (hunk 28 above)
Comment (Shalini-Ashokan) [reply MAUI/Maps/tile-layer.md:16]: Changed

**Review by vigneshrameshsync — APPROVED** [review]
Fine.

**Pull request merged**